flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/enumerator/MySqlSourceEnumerator.java

```diff
@@ -196,6 +204,10 @@ private void syncWithReaders(int[] subtaskIds, Throwable t) {
                         subtaskId, new FinishedSnapshotSplitsRequestEvent());
             }
         }
+
+        for (int subtaskId : subtaskIds) {
+            context.sendEventToSourceReader(subtaskId, new PleaseRequestSplitEvent());
+        }
```

Comment:
Add a check before send event ? Only the time point we change from `binlog reading suspend` to `snapshot reading running` we need to send `PleaseRequestSplitEvent`.